agent/api/task/task.go

```diff
@@ -777,6 +786,43 @@ func (task *Task) initializeFirelensResource(config *config.Config, resourceFiel
 	return errors.New("unable to initialize firelens resource because there's no firelens container")
 }
 
+func (task *Task) addFirelensContainerDependency() error {
+	var firelensContainer *apicontainer.Container
+	for _, container := range task.Containers {
+		if container.FirelensConfig != nil {
+			firelensContainer = container
+		}
+	}
+
+	if firelensContainer == nil {
+		return errors.New("unable to add firelens container dependency because there's no firelens container")
+	}
+
+	for _, container := range task.Containers {
+		if container.DockerConfig.HostConfig == nil {
+			continue
+		}
+
+		hostConfig := &dockercontainer.HostConfig{}
+		err := json.Unmarshal([]byte(*container.DockerConfig.HostConfig), hostConfig)
+		if err != nil {
+			return errors.Wrapf(err, "unable to decode host config of container %s", container.Name)
+		}
+
+		if hostConfig.LogConfig.Type == firelensDriverName {
+			// If there's no dependency between the app container and the firelens container, make firelens container
+			// start first to be the default behavior by adding a START container depdendency.
+			if !container.DependsOnContainer(firelensContainer.Name) && !firelensContainer.DependsOnContainer(container.Name) {
```

Comment:
Are we going to have this documented in the public doc that recommend people to add this dependency for their task def?

Reply:
i think we will document that if no dependency is specified, we will add a firelens container dependency internally by default. not sure why you think we should recommend people to add this dependency themselves?

Reply:
Cause we have a check that if customer defines a dependency already, we will not add the dependency. So if customers define the wrong dependency, it will still lead to the unexpected behavior. That's why I think in doc we should tell customers what is the right dependency look like or they can just leave it not specified.

Reply:
> That's why I think in doc we should tell customers what is the right dependency look like or they can just leave it not specified.

that sounds good to me. i think we can mention these in the doc